bot/app/__init__.py

```diff
@@ -25,60 +26,136 @@
-		photo_caption = f.read()
-	with open(os.path.join(img_dir, 'next_event.jpg'), 'rb') as p:
-		picture=p.read()
-	bot.send_photo(chat_id=message.chat.id, photo=picture, caption=photo_caption, parse_mode='html')
-
-@bot.message_handler(commands=['message_to_all'])
-def push_message_to_all(message):
-	with open(os.path.join(template_dir, 'test.html'), 'r') as f:
-		reply = f.read()
-	uids = db.all_users()
-	for uid in uids:
-		print(uid[0])
-		if (uid != 123456789) and (uid != 987654321):
-			bot.send_message(chat_id=uid[0], text=reply, parse_mode='html')
+
+	"""/next message handler function
+
+	Upon sending the /next command the event's graphic and all available
+	info from the db are returned.
+	Artists are represented with hyperlinks. By pressing one of those,
+	the artist message handler is triggered
+
+	Arguments:
+		message: telebot's message object
+	"""
+
+	with app.app_context():
+		u_id = message.from_user.id
+	
+		try:
+			next_event = db_objects.get_next_event()
+		except:
+			return render_template('none.html')
+	
+		e_name = next_event.name
+		date = next_event.date
+		time = next_event.time
+		admission = next_event.admission
+		description = next_event.description
+		location = next_event.location
+		photo_id = os.path.join(img_dir, next_event.pic_id)
+		
+		artists_temp = db_objects.get_artists_event(e_name)
+		artists_playing = []
+		
+		for artist in artists_temp:
+			artists_playing.append(artist.name)
+
+		try:
+			bot.send_photo(chat_id=u_id, photo=open(photo_id, 'rb'))
+		except Exception as e:
+			logging.error(e)
+
+		try:
+			bot.send_message(chat_id=u_id, text=render_template('next_event.html', \
+			e_name=e_name, date=date, time=time, admission=admission, location=location, \
+			description=description, artists=artists_playing), parse_mode='html')
+		except Exception as e:
+			return render_template('none.html')
+			logging.error(e)
+
+@bot.message_handler(commands=['artist'])
+def artist(message, name):
+
+	"""/artist message handler function
+
+	Upon sending the /artist command this function queries the db for 
+	the corresponding artist and returns facts about it.
+
+	Arguments:
+		message: telebot's message object
+		name: the artist's name
+	"""
+
+	with app.app_context():
+	
+		u_id = message.user.id
+
+		artist = db_objects.get_artist(name)
+		if not artist:
+			return render_template('none.html')
+
+		a_name = artist.name
+		website = artist.website
+		soundcloud = artist.soundcloud
+		bandcamp = artist.bandcamp
+		bio = artist.bio
+		photo_id = artist.pic_id
+
+
+		try:
+			bot.send_photo(chat_id=u_id, photo=open(photo_id, 'rb'))
+		except Exception as e:
+			logging.error(e)
+
+		try:
+			bot.send_message(chat_id=u_id, text=render_template('artist.html', \
+				a_name=a_name, website=website, soundcloud=soundcloud, bandcamp=bandcamp, bio=bio), \
+				parse_mode='html')
+		except Exception as e:
+			return render_template('none.html')
+			logging.error(e)
```

Comment:
`logging.error(e)` never gets called because you return beforehand

Reply:
fixed